### ISSUE TITLE
Implement Training Pipeline

### ISSUE THREAD

**Issue by Mohid-Abbas** [Mohid-Abbas/cmapss-iot-predictive-pipeline#9]
## Description
Implemented complete training pipeline orchestrating model training and evaluation.

## Implementation Details
- Created run_training function as main training entry point
- Loads preprocessed single-row CSV data for XGBoost
- Trains XGBoost model with configured hyperparameters
- Evaluates model on test set with RMSE and MAE metrics
- Runs 5-fold time series cross-validation
- Prints comprehensive performance metrics
- Saves trained model as pickle file for deployment
- Creates models directory if it doesn't exist
- Provides clear console output for monitoring

## Code Reference
```python
def run_training(
    processed_dir: str = "data/processed",
    model_dir: str = "models",
) -> None:
    X_train, X_test, y_train, y_test = load_single_row_data(processed_dir)

    print("Training XGBoost model...")
    model = train_model(X_train, y_train)

    print("Evaluating model...")
    metrics = evaluate_model(model, X_test, y_test)
    print(f"Test RMSE: {metrics['rmse']}")
    print(f"Test MAE: {metrics['mae']}")

    print("Running TimeSeriesSplit cross-validation...")
    cv_scores = cross_validate_model(X_train, y_train, n_splits=5)
    print(f"CV RMSE scores: {[round(s, 4) for s in cv_scores]}")
    print(f"Mean CV RMSE: {round(np.mean(cv_scores), 4)}")

    os.makedirs(model_dir, exist_ok=True)
    joblib.dump(model, os.path.join(model_dir, "xgb_rul_model.pkl"))
    print(f"Saved model to {model_dir}/xgb_rul_model.pkl")
```

## Files Changed
- `src/train.py`

## Status
✅ Completed
